docs/src/main/paradox/scala/http/routing-dsl/httpApp.md

```diff
@@ -0,0 +1,71 @@
+<a id="http-app"></a>
+# HttpApp Bootstrap
+
+@@@ warning { title="API may change" }
+This is experimental and the API is subjected to change in futures releases of Akka HTTP.
+For further information about this marker, see [The @DoNotInherit and @ApiMayChange markers](http://doc.akka.io/docs/akka/current/common/binary-compatibility-rules.html#The_@DoNotInherit_and_@ApiMayChange_markers)
+in the Akka documentation.
+@@@
+
+@@toc { depth=1 }
+
+## Introduction
+
+The objective of `HttpApp` is to quickly get you started starting a Http Server with just a few lines of code.
+This is accomplished just by extending `HttpApp` and implementing the `route()` method.
+If desired, `HttpApp` provides different hook methods that can be overridden to change its default behavior.
+
+## Minimal Example
+
+The following example shows how succinct is to start a server:
+
+@@snip [HttpAppExampleSpec.scala](../../../../../test/scala/docs/http/scaladsl/HttpAppExampleSpec.scala) { #minimal-routing-example }
+
+Firstly we define an `object` (it can also be a `class`) that extends `HttpApp` and we just implement the routes this server will handle.
+After that, we can start a server just by providing a `host`, `port` and `ServerProperties`. Calling `startSever` blocks the current thread until the server is signaled for termination.
+The default behavior of `HttpApp` is to start a server, and shut it down after `ENTER` is pressed. Once this happens, the server is properly shutdown.
+
+## Reacting to Bind Failures
+
+`HttpApp` provides different hooks that will be called after a successful and unsuccessful bindings. For example, the server
+might not start due to the port being already in use, or because it is a privileged one.
+
+Here you can see an example server that overrides the `postHttpBindingFailure` hook and prints the error to the console (this is also the default behavior)
+
+@@snip [HttpAppExampleSpec.scala](../../../../../test/scala/docs/http/scaladsl/HttpAppExampleSpec.scala) { #failed-binding-example }
+
+So if the port 80 would be already taken by another app, the call to `startServer` returns immediately and the `postHttpBindingFailure` hook will be called.
+
+## Providing your own Actor System
+
+`HttpApp` creates its own `ActorSystem` instance when not provided by the user. This is pretty convenient, but more often than not, 
```

Comment:
I suggest switching to 1st person:

```
`HttpApp` creates its own `ActorSystem` instance when one is not provided.
In case you already created an `ActorSystem` in your application you can
pass it to `startServer` as illustrated in the following example:

@@snip [HttpAppExampleSpec.scala](../../../../../test/scala/docs/http/scaladsl/HttpAppExampleSpec.scala) { #with-actor-system }
```